citesphere/src/main/java/edu/asu/diging/citesphere/web/user/GroupItemsController.java

```diff
@@ -110,4 +116,46 @@ public String show(Authentication authentication, Model model, @PathVariable("zo
         model.addAttribute("breadCrumbs", breadCrumbs);
         return "auth/group/items";
     }
+    
+    @RequestMapping(value = {"/auth/group/{zoteroGroupId}/collections/{collectionId}/items"})
+    public ResponseEntity<String> display(Authentication authentication, Model model,
+            @PathVariable("zoteroGroupId") String groupId,
+            @PathVariable(value = "collectionId", required = false) String collectionId,
+            @RequestParam(defaultValue = "1", required = false, value = "page") String page,
+            @RequestParam(defaultValue = "title", required = false, value = "sort") String sort,
+            @RequestParam(required = false, value = "columns") String[] columns)
+            throws GroupDoesNotExistException {
+
+        Integer pageInt = 1;
+        try {
+            pageInt = new Integer(page);
+        } catch (NumberFormatException ex) {
+            logger.warn("Trying to access invalid page number: " + page);
+        }
+
+        IUser user = (IUser) authentication.getPrincipal();
+        CitationResults results = citationManager.getGroupItems(user, groupId, null, pageInt, sort);
+
+        ICollectionResult object = new CollectionResult();
+        object.setTotal(results.getTotalResults());
+        object.setTotalPages(
+                Math.ceil(new Float(results.getTotalResults()) / new Float(zoteroPageSize)));
+        object.setCurrentPage(pageInt);
+        object.setZoteroGroupId(groupId);
+        object.setCollectionId(collectionId);
+        object.setCitationCollections(
+                collectionManager.getCitationCollections(user, groupId, null, pageInt, "title")
+                        .getCitationCollections());
+        List<ICitation> list = results.getCitations();
+        object.setItems(list);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonResponse = "";
+        try {
+            jsonResponse = objectMapper.writeValueAsString(object);
+        } catch (IOException e) {
+            logger.debug("Unable to process JSON response " + e);
```

Comment:
you probably want to log the exception as such (not as string message, but with a comma (as second argument)). Also, should be logged as error, since this might indicate a more general issue if it can't be written as json).